File: src/main/java/ar/edu/unq/epers/unidad5/model/Personaje.java
package ar.edu.unq.epers.unidad5.model;

import ar.edu.unq.epers.unidad5.model.exception.MuchoPesoException;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.HashSet;
import java.util.Set;

@ToString
@Setter
@Getter
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)

@Document("Personaje")
public class Personaje {

    @Id
    private String id;
    private String nombre;
    private int vida;
    private int pesoMaximo;
    private Set<Item> inventario = new HashSet<>();

    public Personaje(String nombre, int vida, int pesoMaximo) {
        this.nombre = nombre;
        this.vida = vida;
        this.pesoMaximo = pesoMaximo;
    }

    public int getPesoActual() {
        return inventario.stream().mapToInt(Item::getPeso).sum();
    }

    public void recoger(Item item) {
        int pesoActual = getPesoActual();
        if (pesoActual + item.getPeso() > this.pesoMaximo) {
            throw new MuchoPesoException(this, item);
        }
        this.inventario.add(item);
    }
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/model/Personaje.kt
package ar.edu.unq.epers.unidad5.model

import ar.edu.unq.epers.unidad5.model.exception.MuchoPesoException
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
class Personaje {
    @Id
    var id: String? = null
    var nombre: String = ""
    var vida: Int = 0
    var pesoMaximo: Int = 0
    var inventario: MutableSet<Item> = HashSet()

    val pesoActual: Int
        get() = inventario.sumOf { it.peso }

    constructor(nombre: String, vida: Int, pesoMaximo: Int) {
        this.nombre = nombre
        this.vida = vida
        this.pesoMaximo = pesoMaximo
    }

    fun recoger(item: Item) {
        val pesoActual = this.pesoActual
        if (pesoActual + item.peso > this.pesoMaximo) {
            throw MuchoPesoException(this, item)
        }

        this.inventario.add(item)
    }

}